pkg/policy/mutate/validate.go

```diff
@@ -62,11 +64,16 @@ func (m *Mutate) validateForEach(tag string, foreach []kyvernov1.ForEachMutation
 
 			return m.validateNestedForEach(tag, fe.ForEachMutation)
 		}
-
 		psm := fe.GetPatchStrategicMerge()
 		if (fe.PatchesJSON6902 == "" && psm == nil) || (fe.PatchesJSON6902 != "" && psm != nil) {
 			return tag, fmt.Errorf("only one of `patchStrategicMerge` or `patchesJson6902` is allowed")
 		}
+		if fe.List == "request.object.spec.containers" {
```

Comment:
This is missing a check that psm is not nil. As you can see from the line above, we could also be in the case that `patchesJson6902` is used.

Reply:
Done